### PR TITLE
Document tt export/import

### LINKED ISSUE #2561: TT (crud) import
**Epic: [Document TT CLI](https://github.com/tarantool/doc/issues/2553)**

Product: Tarantool CLI
Since: 2.10
Audience/target: dev, admin
Root document: TBA
SME: @ Totktonada

# Details

- [ ] tt import
- [ ] tt crud import

`tt crud` interacts with Tarantool's [CRUD](http://github.com/tarantool/crud) module.

Export options: https://github.com/tarantool/doc/issues/3274

Related dev issues:
https://github.com/tarantool/tt-ee/issues/10
https://github.com/tarantool/tt/issues/24

### LINKED ISSUE #3274: tt (crud) export
**Tracked in**: #3168 
**Related dev. issue(s):** https://github.com/tarantool/tt-ee/issues/18 https://github.com/tarantool/tt-ee/issues/19

**Product:** Tarantool CLI
**PM ref:** https://www.notion.so/tarantool/tt-export-7b1b0b6e3635487da1cc286f8a380296
**Since:** ??
**Audience/target:** dev, admin
**Root document:** tt cli docs
**SME:** @ LeonidVas 

# Details

Add docs on the ``tt-ee crud export`` and ``tt-ee export`` modules

# Definition of done

- [ ] a new reference page about ``tt-ee export`` is added
- [ ] a new reference page about ``tt-ee crud export`` is added

# Example

- https://github.com/tarantool/tt-ee/pull/55
- https://github.com/tarantool/tt-ee/blob/master/doc/examples.rst#working-with-tt-crud-export

```
tt export -u testuser -p foobar localhost:3301 file.csv bands
tt import -u testuser -p foobar localhost:3301 file.csv bands2
```

### PR DESCRIPTION
**Pull request by andreyaksenov**
- Created the https://docs.d.tarantool.io/en/doc/tt-export-import/reference/tooling/tt_cli/export page.
- Created the https://docs.d.tarantool.io/en/doc/tt-export-import/reference/tooling/tt_cli/import page.
- Created the `Interacting with the CRUD module` page: https://docs.d.tarantool.io/en/doc/tt-export-import/reference/tooling/tt_cli/crud/. This page contains links to the `tt export`/`tt import` commands with detailed information.
- Added all commands to the index page: https://docs.d.tarantool.io/en/doc/tt-export-import/reference/tooling/tt_cli/commands/. 
- Added a link to the latest release on GitHub and mentioned ``tt EE`` in the intro: https://docs.d.tarantool.io/en/doc/tt-export-import/reference/tooling/tt_cli/
- Added a note about `tt` in the `EE` package: https://docs.d.tarantool.io/en/doc/tt-export-import/reference/tooling/tt_cli/installation/

### CODE REVIEW

**Review by psergee — APPROVED** [review]

File: doc/reference/tooling/tt_cli/crud.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+..  admonition:: Enterprise Edition
+    :class: fact
+
+    This command is supported by the `Enterprise Edition <https://www.tarantool.io/compare/>`_ only.
+
+..  code-block:: console
+
+    $ tt crud COMMAND [COMMAND_OPTION ...]
+
+``tt crud`` enables you to interact with a cluster using the `CRUD <https://github.com/tarantool/crud>`_ module.
```
Comment (p7nov) [inline doc/reference/tooling/tt_cli/crud.rst]: Non-personal wording looks better in references.

```suggestion
``tt crud`` enables the interaction with a cluster using the `CRUD <https://github.com/tarantool/crud>`_ module.
```

File: doc/reference/tooling/tt_cli/export.rst [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    :class: fact
+
+    This command is supported by the `Enterprise Edition <https://www.tarantool.io/compare/>`_ only.
+
+
+..  code-block:: console
+
+    $ tt [crud] export URI FILE SPACE [EXPORT_OPTION ...]
+
+``tt [crud] export`` exports a space's data to a file.
```
Comment (p7nov) [inline doc/reference/tooling/tt_cli/export.rst:16]: Not sure if possessive is correct for this case (`space's data`).
Maybe rephrase? 

```suggestion
``tt [crud] export`` exports data from a given space to a file.
```

Comment (p7nov) [reply doc/reference/tooling/tt_cli/export.rst:16]: Also, maybe mention the file format in this general description?

File: doc/reference/tooling/tt_cli/export.rst [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+``tt [crud] export`` takes the following arguments:
+
+*   ``URI``: The URI of a router instance if ``crud`` is used. Otherwise, it should specify the URI of a storage.
+*   ``FILE``: The name of a file for storing exported data.
+*   ``SPACE``: The name of a space from which data is exported.
+
+..  NOTE::
+
+    You should have :ref:`read access <authentication-owners_privileges>` to the required space to be able to export data.
```
Comment (p7nov) [inline doc/reference/tooling/tt_cli/export.rst]: It's unclear who is `you` in this sentence. The reader runs `tt` outside of Tarantool.
Shorter wording: `to the required space to be able to export data.` > `to the space to export its data`.

File: doc/reference/tooling/tt_cli/export.rst [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+..  NOTE::
+
+    Exporting isn't supported for the :ref:`interval <index-box_interval>` field type.
```
Comment (p7nov) [inline doc/reference/tooling/tt_cli/export.rst]: What exactly does it mean? Export does not work on spaces with such field? Export works but interval fields are lost? Anything else?

File: doc/reference/tooling/tt_cli/export.rst [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    3,2804,Robert,King,33
+    # ...
+
+
+.. _tt-export-compound-data:
+
+Exporting compound data
+-----------------------
+
+By default, ``tt`` exports empty values for fields containing compound data like arrays or maps.
```
Comment (p7nov) [inline doc/reference/tooling/tt_cli/export.rst]:
```suggestion
By default, ``tt`` exports empty values for fields containing compound data such as arrays or maps.
```

Comment (p7nov) [reply doc/reference/tooling/tt_cli/export.rst]: `like` - similar object, `such as` - actual examples.

File: doc/reference/tooling/tt_cli/export.rst [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+
+.. _tt-export-options:
+
+Options
+-------
+
+..  option:: --batch-queue-size INT
+
+    Specify the maximum number of tuple batches in a queue between a fetch and write threads (the default is ``32``).
```
Comment (p7nov) [inline doc/reference/tooling/tt_cli/export.rst]: `Specify` and similar werbs seem redundant to me in options that have values. Why not just give the definition:
`-batch-queue-size: the maximum number...`?

I understand that they are optional and setting them _specifies_ some parameter (in contrast to undefined), but the shorter form is pretty common: https://cloud.google.com/sdk/gcloud/reference/compute/disks/update#GCLOUD-WIDE-FLAGS

File: doc/reference/tooling/tt_cli/export.rst [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+    Set a format used to export compound values like arrays or maps.
+    By default, ``tt`` exports empty values for fields containing such values.
+
+    Supported formats: ``json``.
+
+    See also: :ref:`Exporting compound data <tt-export-compound-data>`.
+
+..  option:: --header
+
+    Specify whether to add field names in the first row.
```
Comment (p7nov) [inline doc/reference/tooling/tt_cli/export.rst]: Will the short form be enough? 

```suggestion
    Add field names in the first row.
```

File: doc/reference/tooling/tt_cli/import.rst [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+Matching of input and space fields
+----------------------------------
+
+
+.. _tt-import-match-fields-auto:
+
+Automatic matching
+~~~~~~~~~~~~~~~~~~
+
+Suppose, you have the ``customers.csv`` file with a header containing field names in the first row:
```
Comment (p7nov) [inline doc/reference/tooling/tt_cli/import.rst]:
```suggestion
Suppose that you have the ``customers.csv`` file with a header containing field names in the first row:
```

File: doc/reference/tooling/tt_cli/import.rst [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+You can also match fields :ref:`manually <tt-import-match-fields-manual>` if field names in the input file and the target space differ.
+Note that if you're importing data into a cluster, you don't need to specify the ``bucket_id`` field.
+The CRUD module generates ``bucket_id`` values automatically.
+
+.. _tt-import-match-fields-manual:
+
+Manual matching
+~~~~~~~~~~~~~~~
+
+The ``--match`` option enables you to import data by matching field names in the input file and the target space manually.
```
Comment (p7nov) [inline doc/reference/tooling/tt_cli/import.rst]:
```suggestion
The ``--match`` option enables importing data by matching field names in the input file and the target space manually.
```

File: doc/reference/tooling/tt_cli/import.rst [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+
+    $ tt crud import localhost:3301 customers.csv customers \
+                     --on-exist replace
+
+.. _tt-import-parsing-error:
+
+Handling parsing errors
+-----------------------
+
+To skip rows whose data cannot be parsed correctly, you can use the ``--on-error`` option as follows:
```
Comment (p7nov) [inline doc/reference/tooling/tt_cli/import.rst]:
```suggestion
To skip rows whose data cannot be parsed correctly, use the ``--on-error`` option as follows:
```

File: doc/reference/tooling/tt_cli/import.rst [hunk 11]
```diff
@@ -0,0 +1,279 @@
...
+..  option:: --quote STRING
+
+    Set the symbol that defines a quote.
+    For CSV, double quotes are used by default (``"``).
+    The double symbol of this option acts as the escaping symbol within input data.
+
+..  option:: -success STRING
+
+    The name of a file with rows that were imported (the default is ``success``).
+    Overrides the file if it already exists.
```
Comment (p7nov) [inline doc/reference/tooling/tt_cli/import.rst]: Can't understand what exactly this means. Overwrites?

File: doc/reference/tooling/tt_cli/installation.rst [hunk 12]
```diff
@@ -5,6 +5,11 @@ To install the ``tt`` command-line utility, use a package manager -- Yum or
 APT on Linux, or Homebrew on macOS. If you need a specific build, you can build
 ``tt`` from sources.
 
+.. NOTE::
+
+    A Tarantool Enterprise's `release package <https://www.tarantool.io/en/enterprise_doc/setup/#package-contents>`_ includes the `tt` utility extended with additional features like :ref:`importing <tt-crud-import>` and :ref:`exporting <tt-crud-export>` data.
```
Comment (p7nov) [inline doc/reference/tooling/tt_cli/installation.rst]: Maybe put this (or similar) text on the main **tt CLI** page to explain the existence/positioning of tt EE in general?

**Review by p7nov — CHANGES_REQUESTED** [review]
Some edits and questions from my side.
Some comments from `export` are applicable to `import` as well; I didn't duplicate them.

File: doc/reference/tooling/tt_cli/export.rst (hunk 2 above)
Comment (andreyaksenov) [reply doc/reference/tooling/tt_cli/export.rst:16]: > a space's data

I believe, possessive works good for our `space` term. 

> Also, maybe mention the file format in this general description?

Currently, only CSV is supported but other formats may be added in the future. And given that CSV is the only supported format, `for example, CSV` will look a bit weird, too.

File: doc/reference/tooling/tt_cli/export.rst (hunk 3 above)
Comment (andreyaksenov) [reply doc/reference/tooling/tt_cli/export.rst]: Will update to `Read access to the space is required to export its data.`

File: doc/reference/tooling/tt_cli/import.rst (hunk 11 above)
Comment (andreyaksenov) [reply doc/reference/tooling/tt_cli/import.rst]: exactly

File: doc/reference/tooling/tt_cli/export.rst (hunk 4 above)
Comment (andreyaksenov) [reply doc/reference/tooling/tt_cli/export.rst]: I've checked: exporting works, data of this field is exported but looks unreadable and causes an error on an attempt to import it back.

File: doc/reference/tooling/tt_cli/export.rst (hunk 6 above)
Comment (andreyaksenov) [reply doc/reference/tooling/tt_cli/export.rst]: Maybe, will remove in all places.

File: doc/reference/tooling/tt_cli/installation.rst (hunk 12 above)
Comment (andreyaksenov) [reply doc/reference/tooling/tt_cli/installation.rst]: Will add it to the intro

File: doc/reference/tooling/tt_cli/export.rst (hunk 2 above)
Comment (p7nov) [reply doc/reference/tooling/tt_cli/export.rst:16]: To be precise, I would write "to a CSV file". If other format appear later, we'll update the description.

File: doc/reference/tooling/tt_cli/export.rst [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+
+*   ``URI``: The URI of a router instance if ``crud`` is used. Otherwise, it should specify the URI of a storage.
+*   ``FILE``: The name of a file for storing exported data.
+*   ``SPACE``: The name of a space from which data is exported.
+
+..  NOTE::
+
+    :ref:`Read access <authentication-owners_privileges>` to the space is required to export its data.
+
+..  NOTE::
```
Comment (p7nov) [inline doc/reference/tooling/tt_cli/export.rst]: Forgot to write in the first review: this should be `important` or even `warning` because the data loss is possible.

File: doc/reference/tooling/tt_cli/index.rst [hunk 14]
```diff
@@ -3,12 +3,17 @@
...
 
+Latest release on GitHub: :tt-release:`1.3.0`
+
 ``tt`` is a utility that provides a unified command-line interface for managing
 Tarantool-based applications. It covers a wide range of tasks -- from installing
 a specific Tarantool version to managing remote instances and developing applications.
 
 ``tt`` is developed in its own `GitHub repository <https://github.com/tarantool/tt>`_.
 Here you can find its source code, changelog, and releases information.
+There is also the Enterprise version of ``tt`` is available in a
```
Comment (p7nov) [inline doc/reference/tooling/tt_cli/index.rst]: Pls add a blank line to separate the new paragraph.

**Review by p7nov — APPROVED** [review]
LGTM, just a couple more notes.
